src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -297,28 +300,39 @@ private constructor(
 
     @RequiresEdt
     fun createFromState(project: Project, state: ChatState): AgentChatSession {
-      val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val modelFromState = state.model?.let { ChatModel.fromDisplayName(it) }
-      val chatSession = AgentChatSession(project, sessionId, state.internalId!!, modelFromState)
-      state.messages.forEachIndexed { index, message ->
-        val parsed =
-            when (val speaker = message.speaker) {
-              MessageState.SpeakerState.HUMAN -> Speaker.HUMAN
-              MessageState.SpeakerState.ASSISTANT -> Speaker.ASSISTANT
-              else -> error("unrecognized speaker $speaker")
-            }
-        val chatMessage = ChatMessage(speaker = parsed, message.text)
-        chatSession.messages.add(chatMessage)
-        chatSession.chatPanel.addOrUpdateMessage(
-            chatMessage, index, shouldAddBlinkingCursor = false)
-      }
+      val agentChatSession = CompletableFuture<AgentChatSession>()
+      createNewPanel(project) { codyAgent ->
+        val innerSessionId = codyAgent.server.chatNew()
+        val mySessionId = innerSessionId.completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
+        val chatModels = codyAgent.server.chatModels(ChatModelsParams(mySessionId))
+        val response = chatModels.completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
+        val modelFromState =
+            response.models.singleOrNull { provider -> provider.model == state.model }
```

Comment:
what is in the `state` on the very first start of the Cody (new user)?

Reply:
I guess there is no such option. This is `createFromState` method.